### PR TITLE
Add first n primes generator

### LINKED ISSUE #19118: Generating first n prime numbers
In the `ntheory.generate` module, there seems to be no straightforward way to generate **the first _n_** primes, equivalent to `primerange(2, prime(n) + 1)`.

In contrast, the `primorial()` function supports computing the product of first _n_ primes as well as the product of primes less than or equal to _n_.

### PR DESCRIPTION
**Pull request by ritesh99rakesh**
#### References to other Issues or PRs

Fixes #19118 

#### Brief description of what is fixed or changed
Added first `n` prime generator - `primes` in `sympy/ntheory/generate.py`. It provides the following functionality
```python
>>> from sympy import primes, Symbol
>>> print([i for i in primes(10)])
[2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
```

#### Other comments
If you feel any other name for the function is better, please suggest.
I will add tests once the concept gets a green flag.
Please review @sylee957 

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by FrackeR011** [comment]
This looks good to me. The name `primes` is very similar to the function `prime` and may cause confusion.

**Comment by sylee957** [comment]
How is this different than `primerange` with sufficiently large `b` or calling `nextprime` repeatedly? Is there any more efficient method available for this when defining the API to be?

File: sympy/ntheory/generate.py [hunk 1]
```diff
@@ -721,6 +721,39 @@ def primerange(a, b):
...
+        This function is equivalent to primerange(2, prime(n) + 1)
+
+        If the range exists in the default sieve, the values will
+        be returned from there; otherwise values will be returned
+        but will not modify the sieve.
+
+        Examples
+        ========
+
+        >>> from sympy import primes, Symbol
```
Comment (iammosespaulr) [inline sympy/ntheory/generate.py:736]:
```suggestion
        >>> from sympy import primes
```

**Comment by czgdp1807** [comment]
Closing this PR as of now due to inactivity and no objections to `Could Close` label. However, if you feel that this PR should be re-opened and finally merged into master, please let us know below. If you have the access to re-open it and continue working on it. Thank you.

**Pull request closed without merging**